Main screen v 1.0

DimaTimofeychik opened the pull request:
Added:
TapBar
The main list of ads

DimaTimofeychik committed:
Main screen v 1.0

mishamorochko commented on AVBYHomework/MainListAnnouncement/IconCollectionViewCell.swift:
```diff
@@ -0,0 +1,12 @@
+final class IconCollectionViewCell: UICollectionViewCell {
+    
+    
+    @IBOutlet weak var iconImageView: UIImageView!
```
private

mishamorochko commented on AVBYHomework/MainListAnnouncement/MainScreenTableViewCell.swift:
```diff
@@ -0,0 +1,140 @@
+    func payPerMonthTapped()
+}
+
+class MainScreenTableViewCell: UITableViewCell {
```
final

mishamorochko commented on AVBYHomework/MainListAnnouncement/MainScreenTableViewCell.swift:
```diff
@@ -0,0 +1,140 @@
+    
+    //MARK: - Outlets
+    
+    @IBOutlet weak var containerView: UIView!
```
private

mishamorochko commented on AVBYHomework/MainListAnnouncement/MainScreenTableViewCell.swift:
```diff
@@ -0,0 +1,140 @@
+    
+    //MARK: - Actions
+    
+    @IBAction func hideButtonTapped(_ sender: Any) {
```
private

mishamorochko commented on AVBYHomework/MainListAnnouncement/MainScreenTableViewCell.swift:
```diff
@@ -0,0 +1,140 @@
+        carPhotosCollectionView.delegate = self
+        carPhotosCollectionView.register(UINib(nibName: "IconCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "IconCollectionViewCell")
+        containerView.layer.cornerRadius = 10
+        //carPhotosCollectionView.layer.cornerRadius = 10
```
Закомментированный код надо удалять

mishamorochko commented on AVBYHomework/MainListAnnouncement/MainScreenTableViewCell.swift:
```diff
@@ -0,0 +1,140 @@
+        //carPhotosCollectionView.layer.cornerRadius = 10
+    }
+    
+    func infoAboutCar(car: CarInfo, indexPath: IndexPath, photos: [UIImage]) {
```
Странное название, если тут передает модельку в ячейку, то лучше setInfoAboutCar(_ car...)

mishamorochko commented on AVBYHomework/MainListAnnouncement/MainScreenTableViewCell.swift:
```diff
@@ -0,0 +1,140 @@
+        self.carPriceInDollarsLabel.text = "≈ " + String(car.carPriceInDollars) + "$"
+        self.carPriceInRublesLabel.text = String(car.carPriceInRubles) + "р."
+        self.carAboutLabel.text = descriptionString(car: car)
+        self.payPerMonthLabel.setTitle("от " + monthlyPayment(car: car, years: 5) + "$/мес", for: .normal)
```
И self. тут можно убрать везде

mishamorochko commented on AVBYHomework/MainListAnnouncement/MainScreenTableViewCell.swift:
```diff
@@ -0,0 +1,140 @@
+        return UICollectionViewCell()
+    }
+    
+//    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
```
Удалить

mishamorochko commented on AVBYHomework/ViewController.swift:
```diff
@@ -1,19 +1,244 @@
 import UIKit
 
 class ViewController: UIViewController {
+    
```
final?

mishamorochko commented on AVBYHomework/ViewController.swift:
```diff
@@ -1,19 +1,244 @@
+    
+    @IBOutlet weak var mainScreenTableView: UITableView!
+    
+    let cars: [CarInfo] = [
```
private

mishamorochko commented on AVBYHomework/ViewController.swift:
```diff
@@ -1,19 +1,244 @@
 class ViewController: UIViewController {
+    
+    
+    @IBOutlet weak var mainScreenTableView: UITableView!
```
private

mishamorochko reviewed (changes requested):
Все ок, нужно исправить некоторые моменты и добавить скриншоты

DimaTimofeychik committed:
Main screen v1.1

Added:
Dark theme support
Top/vin
Fixed the display of photos in collection view
Fixed comments from the past pull&request
I can't fix some UI bugs
